PresnyaHedonist/Controllers/DetailsVC.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  DetailsVC.swift
+//  PresnyaHedonist
+//
+//  Created by Alexandr L. on 7/13/21.
+//
+
+import UIKit
+
+class DetailsVC: UIViewController {
+    
+    var place: Place?
+    
+    @IBOutlet weak var imageView: UIImageView!
```

Comment:
I think I remember seeing a preference for using 'strong' IBOutlets - suggested by Apple themselves. https://www.zerotoappstore.com/should-iboutlets-be-weak-or-strong.html

Reply:
Yes, I read this article too) But honestly, I'm afraid of slow performance...in the case of bigger App of course) 
Could you specify please, what type of outlets really need to be strong? As I understood - all UI elements? 
For now, corrected all Outlets to strong 😇

Reply:
Ok, so you shouldn't be worried about slow performance in this situation.

Capturing something strongly only means: The captured instance will not suddenly disappear while the capturing instance (the view controller) is alive. When the view controller is de-initialized, then all strongly referenced variables will have the reference removed - meaning that if no one else has a strong handle to them, they will disappear too.

When a view is in the view hierarchy it will also be strongly captured by it's parent view - and that's why you are basically seeing that your code is working fine.
But if you were to change your mind later and do some manual manipulation of the view hierarchy, then the references to the views may go away and the view may be denitialized - meaning that the next time you try to update the view through the implicitly unwrapped optional (the ! at the end), you can get a crash.

You should of course worry about 'retain cycles' - but not between a view controller and it's views. Here the ownership direction is pretty clear - the view controller 'owns' the views and can thus capture them strongly without a problem.

The places you could get reference cycles is if the ownership is not modelled correctly. For instance your view controller could be the delegate of one of it's views. It's still correct that the reference from the view controller to the view should be strong, but the delegate reference in the other direction must be weak (for instance, a scroll view shouldn't capture it's viewcontroller strongly when setting the viewcontroller as the delegate of the scroll view).
In that example, the scrollview is created by Apple, so you can rest assured that they already made the delegate property 'weak'.

But if you model your own component that can 'delegate' something back to it's owner, then that delegate should be weak.

There are a few other places in Swift where you need to consider reference cycles but I think that if you learn the delegate pattern, that's a good start.

If you ask yourself: Should this class own this variable, usually you would be able to come up with a pretty clear answer. If not, we could definitely go through such situations and discuss them when you meet them.

Let me know if this is understandible or if I am making a mess of explaining it. ;-)